app/controllers/line_items_controller.rb

```diff
@@ -1,2 +1,52 @@
 class LineItemsController < ApplicationController
+  before_action :find_item, only: %i[ destroy add_quantity reduce_quantity]
+
+  def create       
+    chosen_product = Product.find(params[:product_id])
+    current_cart = @current_cart
+    if current_cart.products.include?(chosen_product)
+      @line_item = current_cart.line_items.find_by(:product_id => chosen_product)
+      @line_item.quantity += 1
+    else
+      @line_item = LineItem.new
+      @line_item.cart = current_cart
+      @line_item.product = chosen_product
+      @line_item.quantity = 1      
+    end
+    @line_item.save!
```

Comment:
@amitshethe save! raises exception if any error occurred.
Handle if any error occurred it should show the error.

Reply:
Handled the exception if any comes

Reply:
I don't see above change